plugins/network-elements/nuage-vsp/src/com/cloud/network/element/NuageVspElement.java

```diff
@@ -196,6 +227,20 @@ public boolean implement(Network network, NetworkOffering offering, DeployDestin
             ResourceUnavailableException, InsufficientCapacityException {
         s_logger.debug("Entering NuageElement implement function for network " + network.getDisplayText() + " (state " + network.getState() + ")");
 
+        if (network.getVpcId() != null) {
+            List<NetworkACLItemVO> rules;
+            if (network.getNetworkACLId() != null) {
+                rules = _networkACLItemDao.listByACL(network.getNetworkACLId());
+            } else {
+                rules = new ArrayList<NetworkACLItemVO>(1);
```

Comment:
Why is a ``List`` of length 1 being created here?  If it should be empty, please use ``Collections.emptyList()``.  Otherwise, it feels like a leaking abstraction.